File: Polyscript/PathAccessNode.cpp
#include "PathAccessNode.h"

using namespace std;

PathAccessNode::PathAccessNode(const vector<Token>& path, Position start, Position end) : Node(start,end)
{
	for (auto item : path) {
		this->path.push_back(item.GetValue().value());
	}
}

vector<string> PathAccessNode::GetPath() const
{
	return path;
}

File: src/parser/nodes/path_access_node.rs
use crate::shared::{postion::Postion, token::Token};

#[derive(Clone, Debug)]
pub struct PathAccessNode {
    pub start: Postion,
    pub end: Postion,
    pub path: Vec<String>
}
impl PartialEq for PathAccessNode  {
    fn eq(&self, _other: &Self) -> bool {
        false
    }   
}
impl PathAccessNode {
    pub fn new(path: Vec<Token>, start: Postion, end: Postion) -> Self {
        let mut items: Vec<String> = vec![];
        for elems in path {
            items.push(elems.value.unwrap().clone());
        }
        Self {
            start,
            end,
            path: items
        }
    }
}